File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.py
class Solution(object):
    def findLeastNumOfUniqueInts(self, arr, k):
        freq = sorted(Counter(arr).values())
        unique = len(freq)

        for f in freq:
            if k >= f:
                k -= f
                unique -= 1
            else:
                break

        return unique

File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.cpp

class Solution {
public:
    unordered_map<int,int> count;
    

    void findCount(vector<int>& arr){
        for (auto c : arr){
            count[c]++;
        }
    }

    int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
        findCount(arr);
        priority_queue<int , vector<int> , greater<int>> pq; 

        for (auto& c : count){
            pq.push(c.second);
        }

        while (!pq.empty() && pq.top() <= k ){
            k-=pq.top();
            pq.pop();
        }

        return pq.size();

    }       
};